### DISCUSSION TITLE
Clarification of minimum order quantity / default order note for a supplier

### DISCUSSION THREAD

**Discussion by twothreenine** [foodcoops/foodsoft#1243] [Decisions]
See discussion in #838 

Problem: `min_order_quantity` was originally intended to be just a textual description and is therefore a string. This is confusing to users, some don't know whether to enter `25 Euro`, `25 €`, or just `25`, or they enter something like `20 kg`.
Later, a very useful feature to only auto-send the order to the supplier if the minimum order quantity is fulfilled was added, which compares this string to the order sum (I believe it is the gross sum, but this should be clarified in a tooltip as well).

My suggestion:
1. Add a text field `supplier.order_note` which prefills the `order.note` field of each order opened for this supplier. This could be used for various information like minimum order quantity in crates etc., transport cost, link to supplier's website etc. (Additionally, we could also extend the `custom fields` feature to orders, but that would take more work.)
2. Then we could migrate `min_order_quantity` to a monetary field (rename it to `min_order_value`?), detect all non-numeric / non-monetary data saved in it, and move it into `supplier.order_note` (with a label `Minimum order quantity: ` in the respective locale). E.g. `25 Euro` or `25 €` should be converted to `25`, but `20 kg` should be moved to `supplier.order_note`.
3. Add currency symbol and clarification tooltip to minimum order value.

**Comment by RayOei** [comment]
@twothreenine Looking at this issue and the recent reported #1166 I think this should work without the need of tool tips (no problem having them though) so users can't mess up.

Another option, without reworking the structure as such could be changing `app.models.order.rb line 324` 
I am not sure why the `to_r` is being used as this may lead to issues with the comparison, as is reported in #1166 eg. `sum=49` with `min_order_quantity = "€ 50,00"` will return `true` which is incorrect as the input is interpreted as `0`.
 Would the input have been `50€` it would be processed as `50`, and the comparison would return `false` (as would be expected).

So changing
```
send_to_supplier!(created_by) if sum >= supplier.min_order_quantity.to_r
```

to
```
send_to_supplier!(created_by) if sum >= supplier.min_order_quantity.delete("^0-9,.").to_i
```

will filter unwanted characters and protect users who have added anything like `€ 50`, `€ 50.00`, `€ 50,00`.
It will truncate anything decimal, though.

A more overall consistent solution would be preferable, obviously.

**Comment by twothreenine** [comment]
It would be awesome if we could solve this for the 5.0 release, so we can differentiate between the terms _minimum order value_ (monetary, for a supplier) and _minimum order quantity_ (in unit amount, for a certain article).
That's why I'll give it the extended unit PR label.

**Comment by lentschi** [comment]
@twothreenine I agree that this is an issue and think the [approach you provided to solve it](https://github.com/foodcoops/foodsoft/discussions/1243#discussion-9114430) sounds great. :+1: 

> It would be awesome if we could solve this for the 5.0 release

It would be awesome, however I don't think it should block the release. (It's completely unrelated to the article unit change as this is about money only and we didn't change anything here. Or am I missing something?)

> That's why I'll give it the extended unit PR label.

For me, this label was never meant to mark issues/discussions that should be _in the release_. It's just a remnant intended to mark those issues we moved here from our "hackathon" fork (Sometimes the comments were referring to internal fork topics and the label should warn readers about that.)

@RayOei 
Concerning [your workaround solution](https://github.com/foodcoops/foodsoft/discussions/1243#discussioncomment-14895876). Yes that would probably enhance the current code somewhat, but I'd prefer going straight for Leo's clean solution instead, if you don't mind?

@twothreenine If there's no further comments, I think you could turn this into an issue right away.
